src/maidsafe/vault/mpid_manager/service.cc

```diff
@@ -344,18 +346,140 @@ void MpidManagerService::HandleMessage(
   }
 }
 
+void MpidManagerService::HandleChurnEvent(
+  std::shared_ptr<routing::CloseNodesChange> close_nodes_change) {
+  try {
+    std::lock_guard<std::mutex> lock(mutex_);
+    if (stopped_)
+      return;
+    close_nodes_change_ = *close_nodes_change;
+    //    VLOG(VisualiserAction::kConnectionMap, close_nodes_change->ReportConnection());
+    MpidManager::TransferInfo transfer_info(handler_.GetTransferInfo(close_nodes_change));
+    for (const auto& transfer : transfer_info)
+      TransferAccount(transfer.first, transfer.second);
+  }
+  catch (const std::exception& e) {
+    LOG(kError) << "Error : " << boost::diagnostic_information(e) << "\n\n";
+  }
+}
+
+void MpidManagerService::TransferAccount(const NodeId& destination,
+                                         const std::vector<MpidManager::KVPair>& accounts) {
+  assert(!accounts.empty());
+  protobuf::AccountTransfer account_transfer_proto;
+  {
+//    std::lock_guard<std::mutex> lock(mutex_);
```

Comment:
If we don't need the lock guard (which it looks like we don't) could this line be removed please?  It looks suspicious and will keep making reviewers or maintainers requestion whether there should be a lock here.